src/components/map/Layers/Countries.tsx

```diff
@@ -1,133 +1,156 @@
+        {
+            source: COUNTRY_LAYER_ID,
+            sourceLayer: COUNTRY_LAYER_ID,
+            id: iso3Name,
+        },
+        {
+            isHighlighted: true
+        }
+    );
 }
 
-const Countries = (map: mapboxgl.Map | undefined, estimateGradePrevalences: EstimateGradePrevalence[]) => {
+const Countries = (map: mapboxgl.Map | undefined, {estimateGradePrevalences, countryFocus}: CountriesMapConfig) => {
 
     const [state] = useContext(AppContext);
     const [popup, setPopup] = useState<mapboxgl.Popup | undefined>(undefined);
-    const [currentFeatureId, setCurrentFeatureId] = useState<string | undefined>(undefined);
+    const [highlight, setHighlight] = useState<string | undefined>(undefined);
+    const history = useHistory();
 
     // If estimates are updated, waits until map is loaded then maps estimate data to country features
     useEffect(() => {
-        if (estimateGradePrevalences.length > 0 && map && map.getSource('Countries')) {
-            SetCountryEstimates(map, estimateGradePrevalences);
-        }
-        else if (map) {
-            map.on('sourcedata', function (e: any) {
-                if (e.sourceId === 'Countries' && map.isSourceLoaded('Countries')) {
+
+        if (map)
+        {
+            if(estimateGradePrevalences)
+            {
+                const sourceLoaded = map.getSource(COUNTRY_LAYER_ID) !== undefined
+                if (!sourceLoaded) 
+                {
+                    map.on('styledata', ()=>SetCountryEstimates(map, estimateGradePrevalences));
+                }
+                else {
                     SetCountryEstimates(map, estimateGradePrevalences);
                 }
-            });
+            }
+            else if(countryFocus)
+            {
+                SetStudiesMode(map, countryFocus);
+            }
         }
-    }, [estimateGradePrevalences, map]);
+    }, [map, estimateGradePrevalences, countryFocus]);
 
-    // If estimates are updated, waits until map is loaded then filters country features
-    useEffect(() => {
-        if (estimateGradePrevalences.length > 0 && map) {
-            FilterCountryEstimates(map, estimateGradePrevalences)
-        }
-    }, [estimateGradePrevalences, map]);
 
     // wait until map is loaded then creates and binds popup to map events
     useEffect(() => {
         if (map && popup === undefined) {
-            const countryPop = new mapboxgl.Popup({ offset: 25, className: "pin-popup", closeButton: false });
-
-            map.on("mouseenter", "Countries", function (e: any) {
-                if (e.features[0].state.hasData) {
-                    countryPop.setMaxWidth("250px")
-                        .setHTML(ReactDOMServer.renderToString(CountryPopup(e.features[0], state.language)))
-                        .setLngLat(e.lngLat)
-                        //.trackPointer()
-                        .addTo(map);
-                }
+            const countryPopup = new mapboxgl.Popup({
```

Comment:
can this be typed?

Reply:
I believe its implicitly typed from mapbox's own popup type